Mod1/Lessons/Week5/Scope.md

```diff
@@ -0,0 +1,232 @@
+```
+
+You're teammate Sergio wrote the code below using Isabella's BankAccount class. It's your job to give Sergio feedback on his code. Take 5 minutes and discuss with your partner what feedback you would give Sergio. How is Sergio using the BankAccount class differently than Isabella expected he would? Why is this a problem?
+
+```c#
+class MisusingBankAccountClass
+{
+    static void Main()
+    {
+        var bankAccountGiana = new BankAccount("Giana");
+        bankAccountGiana.Balance += 100;
+        Console.WriteLine(bankAccountGiana.Balance);
+    }
+}
+```
+
+
+Did you see how Sergio is directly updating the balance of the account? He isn't using the MakeDeposit method that Isabella created. And because he's not using the MakeDeposit method, the code to update the records for the deposit isn't run!
+
+
+Isabella expected other developers would use the BankAccount class as follows where they use methods instead of modifying the balance directly.
+
+```c#
+class UsingBankAccountClass
+{
+    static void Main()
+    {
+        var bankAccountGiana = new BankAccount("Giana");
+        bankAccountGiana.MakeDeposit(100);
+        Console.WriteLine(bankAccountGiana.GetBalance());
+    }
+}
+```
+
+Is there a way to structure the BankAccount class to ensure other developers used the methods as expected and don't directly modify the balance?
+
+This is where access modifiers come in!
+
+Instead of defining the Balance variable as `public` Isabella could have defined it as `private`. This would have caused Sergio to get the following error.
+
+
+![Access Modifiers Example](/Mod1/Images/Week5/AccessModifiersExample.png)
+
+## Definitions
+
+`public`: The code is accessible by all classes
+
+`private`: The code is only accessible within the same class
+
+Access modifiers can be applied to both variables and methods. Note above how Balance is still available within the MakeDeposit method because the MakeDeposit method is in the same class.
+
+
+It can be helpful to as yourself the following questions when determining `public` vs `private`.
+
+Do I need code outside of the class to have access to this variable or method? (if yes, make it public)
+
+Is this an internal detail that should not be accessed from outside of the class? (if yes, make it private)
+
+<!-- Not sure we should talk about how hiding implementation details is an OOP pattern, encapsalation. Or leave that for a later day and reference public vs private as an example. -->
+
+You *could* make all of your variables and methods public and the code would still work, but as we saw, that can lead to unexpected uses of the class and headaches down the line.
+
+C# provides other types of access modifiers for more specific situations such as protected and internal. If you find yourself wanting a level of access between public and private, you can learn more in the documentation [here](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/access-modifiers). 
+
```

Comment:
I think this overview of public/private is great - it focuses on the definitions without going into too much discussion of nuance.  I think Mod1 students will be able to pick up this concept and build on it.